File: backend/internal/models/ai.go
package models

type AIModelConfigs struct {
	ChatModel AIModelConfig
}

type AIModelConfig struct {
	APIKey    string
	BaseURL   string
	ModelName string
}

type AITextStreamingCallback func(chunk []byte) error

// GenerateText structs
type AIGenerateTextRequest struct {
	Prompt string `json:"prompt" binding:"required"`
}

type AIGenerateTextResponse struct {
	Content string `json:"content"`
}

// GenerateTextContentOptimizationRequest structs
type AIGenerateTextContentOptimizationRequest struct {
	Context string `json:"context" binding:"required"`
	Style   string `json:"style" binding:"required"`
}

type AIGenerateTextContentOptimizationResponse struct {
	Content string `json:"content"`
}

// GenerateTextCreatePostContentRequest structs
type AIGenerateTextCreatePostContentRequest struct {
	Topic string `json:"topic" binding:"required"`
	Style string `json:"style" binding:"required"`
}

type AIGenerateTextCreatePostContentResponse struct {
	Content string `json:"content"`
}

File: frontend/src/apis/models/ai.ts
export interface ErrorResponse {
  error: string;
}

export interface AIGenerateTextCreatePostContentRequest {
  topic: string;
  style: string;
}

export interface AIGenerateTextCreatePostContentResponse {
  content: string;
}

export interface AIGenerateTextContentOptimizationRequest {
  context: string;
  style: string;
}

export interface AIGenerateTextContentOptimizationResponse {
  content: string;
}